spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/handler/predicate/CloudFoundryRouteServicePredicateFactory.java

```diff
@@ -0,0 +1,44 @@
+package org.springframework.cloud.gateway.handler.predicate;
+
+import java.util.function.Predicate;
+
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * Creates a predicate which indicates if the request is intended for a Cloud Foundry Route Service.
+ * @see <a href="https://docs.cloudfoundry.org/services/route-services.html">Cloud Foundry Route Service documentation</a>.
+ * @author Andrew Fitzgerald
+ */
+public class CloudFoundryRouteServicePredicateFactory extends
+		AbstractRoutePredicateFactory<CloudFoundryRouteServicePredicateFactory.Config> {
+
+	public static final String X_CF_FORWARDED_URL = "X-CF-Forwarded-Url";
+	public static final String X_CF_PROXY_SIGNATURE = "X-CF-Proxy-Signature";
+	public static final String X_CF_PROXY_METADATA = "X-CF-Proxy-Metadata";
+	private final HeaderRoutePredicateFactory factory = new HeaderRoutePredicateFactory();
+
+	public CloudFoundryRouteServicePredicateFactory() {
+		super(CloudFoundryRouteServicePredicateFactory.Config.class);
+	}
+
+	@Override
+	public Predicate<ServerWebExchange> apply(
+			CloudFoundryRouteServicePredicateFactory.Config unused) {
+		return headerPredicate(X_CF_FORWARDED_URL)
+				.and(headerPredicate(X_CF_PROXY_SIGNATURE))
+				.and(headerPredicate(X_CF_PROXY_METADATA));
+	}
+
+	private Predicate<ServerWebExchange> headerPredicate(String header) {
+		HeaderRoutePredicateFactory.Config config = factory.newConfig();
+		config.setHeader(header);
+		config.setRegexp(".*");
+		return factory.apply(config);
+	}
+
+	@Validated
+	public static class Config {
```

Comment:
I believe you can leave this out.

Reply:
What should I do about the `AbstractRoutePredicateFactory` type parameter?

Reply:
Object.class. There's a shortcut constructor in the abstract filter.